dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java

```diff
@@ -719,27 +840,34 @@ protected Ds3Metadata call() throws Exception {
                 final Ds3Client client = session.getClient();
                 final Ds3TreeTableValue value = values.get(0).getValue();
                 final HeadObjectResponse headObjectResponse = client.headObject(new HeadObjectRequest(value.getBucketName(), value.getFullName()));
-                return new Ds3Metadata(headObjectResponse.getMetadata(), headObjectResponse.getObjectSize(), value.getFullName());
+                return new Ds3Metadata(headObjectResponse.getMetadata(), headObjectResponse.getObjectSize(), value.getFullName(), value.getLastModified());
             }
         };
         workers.execute(getMetadata);
         getMetadata.setOnSucceeded(event -> Platform.runLater(() -> {
             LOG.info("Launching metadata popup");
-            MetadataPopup.show(getMetadata.getValue());
+            final MetadataView metadataView = new MetadataView(getMetadata.getValue());
+            Popup.show(metadataView.getView(), resourceBundle.getString("metaDataContextMenu"));
         }));
     }
 
     private void deleteFolderPrompt() {
         LOG.info("Got delete folder event");
-        final ImmutableList<TreeItem<Ds3TreeTableValue>> values = ds3TreeTable.getSelectionModel().getSelectedItems().stream().collect(GuavaCollectors.immutableList());
-        if (values.isEmpty()) {
+        ImmutableList<TreeItem<Ds3TreeTableValue>> values = ds3TreeTable.getSelectionModel().getSelectedItems().stream().collect(GuavaCollectors.immutableList());
+        final TreeItem<Ds3TreeTableValue> root = ds3TreeTable.getRoot();
+        if (values.isEmpty() && null == root) {
             LOG.error("No files selected");
             return;
+        } else if (values.isEmpty()) {
+            final ImmutableList.Builder<TreeItem<Ds3TreeTableValue>> builder = ImmutableList.builder();
+            values = builder.add(root).build().asList();
```

Comment:
If this is just adding a single value this can be simplified to:

```java

values = ImmutableList.of(root);
```

I'm also not sure that the `asList()` call is necessary.